TicketsDemo.Mongo/Repositories/MongoTrainRepository.cs

```diff
@@ -0,0 +1,65 @@
+﻿using MongoDB.Driver;
+using System.Collections.Generic;
+using System.Linq;
+using MongoDB.Bson;
+using TicketsDemo.Data;
+using TicketsDemo.Data.Repositories;
+using TicketsDemo.Data.Entities;
+
+
+namespace TicketsDemo.Mongo
+{
+    public class MongoTrainRepository: ITrainRepository
+    {
+        public List<Train> GetAllTrains()
+        {
+            var ctx = new TContext();
+            var trains = ctx.Trains.Find(new BsonDocument()).ToList();
+            foreach (var train in trains)
+            {
+                Init_Train(train);
+            }
+            return trains;
+        }
+
+        public Train GetTrainDetails(int id)
+        {
+            var ctx = new TContext();
+            var train = ctx.Trains.Find(new BsonDocument("_id", id)).FirstOrDefaultAsync().Result;
+            Init_Train(train);
+            return train;
+        }
+
+        public void CreateTrain(Data.Entities.Train train)
+        {
+            var ctx = new TContext();
+            ctx.Trains.InsertOneAsync(train);
```

Comment:
why async? If you are inserting it 'Async' - you should await action's completion, shouldn't you?